File: pmb/ai_manage/models/calendar_event.py
"""纪念日历数据模型"""
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class MemorialEvent:
    date: str                # 纪念日期 "YYYY-MM-DD"
    event_type: str          # 事件类型: milestone/life_change/major_purchase/emotion/growth
    title: str               # 事件标题
    description: str         # 温馨文案
    related_transactions: list[dict] = field(default_factory=list)  # 关联交易记录
    importance: int = 5      # 重要性评分 1-10

    def to_dict(self) -> dict:
        return {
            "date": self.date,
            "event_type": self.event_type,
            "title": self.title,
            "description": self.description,
            "related_transactions": self.related_transactions,
            "importance": self.importance,
        }

    @classmethod
    def from_dict(cls, d: dict) -> "MemorialEvent":
        return cls(
            date=d.get("date", ""),
            event_type=d.get("event_type", ""),
            title=d.get("title", ""),
            description=d.get("description", ""),
            related_transactions=d.get("related_transactions", []),
            importance=d.get("importance", 5),
        )


@dataclass
class MemorialCalendar:
    user_name: str
    events: list[MemorialEvent] = field(default_factory=list)
    generated_at: str = field(default_factory=lambda: datetime.now().isoformat())
    model_used: str = ""

    def to_dict(self) -> dict:
        return {
            "user_name": self.user_name,
            "events": [e.to_dict() for e in self.events],
            "generated_at": self.generated_at,
            "model_used": self.model_used,
        }

    @classmethod
    def from_dict(cls, d: dict) -> "MemorialCalendar":
        return cls(
            user_name=d.get("user_name", ""),
            events=[MemorialEvent.from_dict(e) for e in d.get("events", [])],
            generated_at=d.get("generated_at", ""),
            model_used=d.get("model_used", ""),
        )

File: aipmb_manage/lib/models/calendar_event.dart
class MemorialEvent {
  final String date;
  final String eventType;
  final String title;
  final String description;
  final List<Map<String, dynamic>> relatedTransactions;
  final int importance;

  MemorialEvent({
    required this.date,
    required this.eventType,
    required this.title,
    required this.description,
    required this.relatedTransactions,
    required this.importance,
  });

  factory MemorialEvent.fromJson(Map<String, dynamic> json) {
    return MemorialEvent(
      date: json['date'] ?? '',
      eventType: json['event_type'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      relatedTransactions: List<Map<String, dynamic>>.from(json['related_transactions'] ?? []),
      importance: json['importance'] ?? 5,
    );
  }

  String get eventTypeLabel {
    switch (eventType) {
      case 'milestone':
        return '里程碑';
      case 'life_change':
        return '生活变迁';
      case 'major_purchase':
        return '重要消费';
      case 'emotion':
        return '情感记忆';
      case 'growth':
        return '成长轨迹';
      default:
        return '纪念';
    }
  }
}

class MemorialCalendar {
  final String userName;
  final List<MemorialEvent> events;
  final String generatedAt;
  final String modelUsed;

  MemorialCalendar({
    required this.userName,
    required this.events,
    required this.generatedAt,
    required this.modelUsed,
  });

  factory MemorialCalendar.fromJson(Map<String, dynamic> json) {
    return MemorialCalendar(
      userName: json['user_name'] ?? '',
      events: (json['events'] as List<dynamic>?)
              ?.map((e) => MemorialEvent.fromJson(e))
              .toList() ??
          [],
      generatedAt: json['generated_at'] ?? '',
      modelUsed: json['model_used'] ?? '',
    );
  }
}